build.sh

```diff
@@ -73,153 +75,186 @@ function create_tempfile {
     fi
 }
 
-if [ ! -d packages ]; then
-    . ./update.sh
-fi
+function create_cpio {
+    local KERNEL_VERSION=""
+    local target_system
 
-rm -rf tmp
-mkdir tmp
+    if [ "$1" = "rpi1" ] ; then
+        KERNEL_VERSION=$KERNEL_VERSION_RPI1
+        target_system="rpi1"
+    elif [ "$1" = "rpi2" ] ; then
+        KERNEL_VERSION=$KERNEL_VERSION_RPI2
+        target_system="rpi2"
+    else
+        echo "Invalid parameter to 'create_cpio' function!"
+        return 1
+    fi
 
-# extract debs
-for i in packages/*.deb; do
-    cd tmp && ar x ../$i && tar -xf data.tar.*; rm data.tar.*; cd ..
-done
+    # initialize rootfs
+    rm -rf rootfs
+    mkdir -p rootfs
+    # create all the directories needed to copy the various components into place
+    mkdir -p rootfs/bin/
+    mkdir -p rootfs/lib/
```

Comment:
this line can be removed, because the next mkdir will create this dir anyway... (or remove the -p params)